gazebo_ros/include/gazebo_ros/node.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef GAZEBO_ROS_NODE_HPP
+#define GAZEBO_ROS_NODE_HPP
+
+#include <memory>
+#include <mutex>
+#include <atomic>
+
+#include <rclcpp/rclcpp.hpp>
+#include <gazebo/common/common.hh>
+
+#include <gazebo_ros/executor.hpp>
+
+namespace gazebo_ros
+{
+
+/// ROS Node for gazebo plugins
+class Node : public rclcpp::Node
+{
+public:
+  /// Exception thrown when a #Load is called for a Node before #InitRos has ever been called
+  class NotInitializedException : public std::runtime_error
+  {
+    public:
+      NotInitializedException();
+  };
+
+  typedef std::shared_ptr<Node> SharedPtr;
```

Comment:
How about making this type more explicit? Such as `NodeSharedPtr`? That should make the API clearer and we don't end up reserving a very general "keyword".

Reply:
This is the convention used in [other parts of the ROS2 codebase](https://github.com/ros2/rclcpp/blob/24f39700c666044aa9cb5dc47dccaf2ae2d93e2e/rclcpp/include/rclcpp/macros.hpp) so far. The typedef is within the class, so it is accessed as ```gazebo_ros::Node::SharedPtr```

Reply:
Ahh got it, good call!